server/shared/src/test/scala/org/http4s/server/middleware/GZipSuite.scala

```diff
@@ -88,6 +89,44 @@ class GZipSuite extends Http4sSuite {
     resp.map(!_.headers.contains[`Content-Encoding`]).assert
   }
 
+  test("decodes random content-type if given isZipped is true") {
+    val request = "Request string"
+    val routes: HttpRoutes[IO] = HttpRoutes.of[IO] { case req @ POST -> Root => Ok(req.body) }
```

Comment:
I didn't know how to check and assert request body here so I figured I might as well return it as a response and assert it later. 
Is there other, preffered way?